cogs/moderation.py

```diff
@@ -51,11 +79,72 @@ async def ban(self, ctx: Context, member: discord.Member, *, reason: str = "No R
         await member.send(embed=embed)
         await member.ban(reason=reason)
 
+    class ActionReason(Converter):
+        async def convert(self, ctx: Context, argument: str) -> str:
+            ret = f"{ctx.author} (ID: {ctx.author.id}): {argument}"
+
+            if len(ret) > 512:
+                reason_max = 512 - len(ret) + len(argument)
+                raise BadArgument(f"Reason is too long ({len(argument)}/{reason_max})")
+            return ret
+
+    class MemberID(Converter):
+        async def convert(self, ctx: Context, argument: str) -> t.Union[str, type]:
+            try:
+                member = await MemberConverter().convert(ctx, argument)
+            except BadArgument:
+                try:
+                    member_id = int(argument, base=10)
+                    member = await resolve_member(ctx.guild, member_id)
+                except ValueError:
+                    raise BadArgument(f"{argument} is not a valid member or member ID.") from None
+                except MemberNotFound:
+                    # hackban case
+                    return type("_Hackban", (), {"id": member_id, "__str__": lambda s: f"Member ID {s.id}"})()
+
+            if not can_execute_action(ctx, ctx.author, member):
+                raise BadArgument("You cannot do this action on this user due to role hierarchy.")
+            return member
```

Comment:
You might want to create a decorator for this instead of checking it in `MemberID` as that doesn't really make sense, this should be a converter not a check.

Reply:
we can create a file for decorators later. pls, approve it for now.